bands/site_sections/bands.py

```diff
@@ -9,128 +9,159 @@ def index(self, session, id, message=''):
 
             if not message:
+                band.bio = band_bio
+                session.add(band)
+                session.add(band_bio)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'Your bio information has been updated')
 
         return {
             'band': band,
             'message': message
         }
 
-    def w9(self, session, id, message='', w9=None):
-        band = session.band(id)
+    def w9(self, session, band_id, message='', w9=None, **params):
+        band = session.band(band_id)
+        band_taxes = session.band_taxes(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            band.w9_filename = w9.filename
-            band.w9_content_type = w9.content_type.value
-            if band.w9_extension not in c.ALLOWED_W9_EXTENSIONS:
+            band_taxes.w9_filename = w9.filename
+            band_taxes.w9_content_type = w9.content_type.value
+            if band_taxes.w9_extension not in c.ALLOWED_W9_EXTENSIONS:
                 message = 'Uploaded file type must be one of ' + ', '.join(c.ALLOWED_W9_EXTENSIONS)
             else:
-                with open(band.w9_fpath, 'wb') as f:
+                with open(band_taxes.w9_fpath, 'wb') as f:
                     shutil.copyfileobj(w9.file, f)
+                band.taxes = band_taxes
+                session.add(band)
+                session.add(band_taxes)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'W9 uploaded')
 
         return {
             'band': band,
             'message': message
         }
 
-    def stage_plot(self, session, id, message='', plot=None):
-        band = session.band(id)
+    def stage_plot(self, session, band_id, message='', plot=None, **params):
+        band = session.band(band_id)
+        band_stage_plot = session.band_stage_plot(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            band.stage_plot_filename = plot.filename
-            band.stage_plot_content_type = plot.content_type.value
-            if band.stage_plot_extension not in c.ALLOWED_STAGE_PLOT_EXTENSIONS:
+            band_stage_plot.filename = plot.filename
+            band_stage_plot.content_type = plot.content_type.value
+            if band_stage_plot.stage_plot_extension not in c.ALLOWED_STAGE_PLOT_EXTENSIONS:
                 message = 'Uploaded file type must be one of ' + ', '.join(c.ALLOWED_STAGE_PLOT_EXTENSIONS)
             else:
-                with open(band.stage_plot_fpath, 'wb') as f:
+                with open(band_stage_plot.fpath, 'wb') as f:
                     shutil.copyfileobj(plot.file, f)
+                band.stage_plot = band_stage_plot
+                session.add(band)
+                session.add(band_stage_plot)
                 raise HTTPRedirect('index?id={}&message={}', band.id, 'Stage directions uploaded')
 
         return {
             'band': band,
             'message': message
         }
```

Comment:
By not passing the stage plot here, you're not going to be able to include the ``.db_id`` (i.e. the current id or ``None`` if this is the first time the form has been filled out) in the form, which means that you won't get the nice "insert or update" behavior you're getting elsewhere.